util/crc32c.cc

```diff
@@ -387,17 +394,41 @@ static bool isSSE42() {
   return false;
 #endif
 }
+#endif
 
 typedef uint32_t (*Function)(uint32_t, const char*, size_t);
 
+#if defined(HAVE_POWER8) && defined(HAS_ALTIVEC) 
+uint32_t ExtendPPCImpl(uint32_t crc, const char* buf, size_t size) {
+  return crc32c_ppc(crc,(const unsigned char*)buf,size);
+}
+
+static bool isAltiVec(){
+  if (arch_ppc_probe()) {
+    return true;
+  }else{
+    return false;
+  }
+}
+# endif
+
 static inline Function Choose_Extend() {
+#ifndef  HAVE_POWER8
   return isSSE42() ? ExtendImpl<Fast_CRC32> : ExtendImpl<Slow_CRC32>;
+#else
+  return isAltiVec() ? ExtendPPCImpl : ExtendImpl<Slow_CRC32>;
+#endif
 }
 
 bool IsFastCrc32Supported() {
```

Comment:
Its use by DumpSupportInfo and potential for future enhancement this probably should return a const char * describing the implementation rather than a bool. Bit outside the scope of the PR however.